src/Microsoft.Health.Dicom.Core/Features/Store/IStoreResponseBuilder.cs

```diff
@@ -21,11 +21,18 @@ public interface IStoreResponseBuilder
         StoreResponse BuildResponse(string studyInstanceUid);
 
         /// <summary>
-        /// Adds a successful entry to the response.
+        /// Adds a Success entry to the response.
         /// </summary>
         /// <param name="dicomDataset">The DICOM dataset that was successfully stored.</param>
         void AddSuccess(DicomDataset dicomDataset);
 
+        /// <summary>
+        /// Adds a Warning entry to the response.
+        /// </summary>
+        /// <param name="dicomDataset">The DICOM dataset that was successfully stored.</param>
+        /// <param name="warningReasonCode">The warning reason code.</param>
+        void AddWarning(DicomDataset dicomDataset, ushort warningReasonCode);
```

Comment:
Just to clarify intent here, what about calling this method `AddSuccessWithWarning`